src/Raven.Server/Documents/AttachmentsStorage.cs

```diff
@@ -901,6 +922,124 @@ public AttachmentDetails MoveAttachment(DocumentsOperationContext context, strin
+                using (GetAttachmentPartialKey(context, lowerId.Content.Ptr, lowerId.Size, lowerName.Content.Ptr, lowerName.Size, AttachmentType.Document, changeVector: null, out Slice partialKeySlice))
+                {
+                    if (table.SeekOnePrimaryKeyPrefix(partialKeySlice, out _) == false)
+                        break;
+
+                    newName = $"{prefix}_#{++count}_{name}";
+                }
+            }
+
+            return newName;
+        }
+
+        public bool DeleteAttachment(DocumentsOperationContext context, Slice lowerId, string name, string hash, string contentType, AttachmentType attachmentType)
+        {
+            var docId = lowerId.ToString();
+            if (string.IsNullOrWhiteSpace(docId))
+                throw new ArgumentException("Argument cannot be null or whitespace.", nameof(lowerId));
+            if (string.IsNullOrWhiteSpace(name))
+                throw new ArgumentException("Argument cannot be null or whitespace.", nameof(name));
+            if (string.IsNullOrWhiteSpace(hash))
+                throw new ArgumentException("Argument cannot be null or whitespace.", nameof(hash));
+            if (string.IsNullOrWhiteSpace(contentType))
+                throw new ArgumentException("Argument cannot be null or whitespace.", nameof(contentType));
+            if (attachmentType != AttachmentType.Document)
+                throw new ArgumentException($"{nameof(DeleteAttachmentDirect)} only supported for documents attachment.", nameof(attachmentType));
+            if (context.Transaction == null)
+                throw new ArgumentException($"Context must be set with a valid transaction before calling {nameof(DeleteAttachmentDirect)}", nameof(context));
+
+            using (DocumentIdWorker.GetLowerIdSliceAndStorageKey(context, name, out Slice lowerName, out Slice namePtr))
+            using (DocumentIdWorker.GetLowerIdSliceAndStorageKey(context, contentType, out Slice lowerContentType, out Slice contentTypePtr))
+            using (Slice.From(context.Allocator, hash, out Slice base64Hash)) // Hash is a base64 string, so this is a special case that we do not need to escape
+            using (GetAttachmentKey(context, lowerId.Content.Ptr, lowerId.Size, lowerName.Content.Ptr, lowerName.Size, base64Hash, lowerContentType.Content.Ptr,
+                lowerContentType.Size, AttachmentType.Document, Slices.Empty, out Slice keySlice))
+            {
+                var table = context.Transaction.InnerTransaction.OpenTable(AttachmentsSchema, AttachmentsMetadataSlice);
+                if (table.SeekOnePrimaryKeyPrefix(keySlice, out TableValueReader tvr) == false)
+                    return false;
+
+                var attachment = TableValueToAttachment(context, ref tvr);
+                if (attachment == null)
+                    AttachmentDoesNotExistException.ThrowFor(docId, name);
```

Comment:
Why are you throwing here?

Reply:
I think we should not throw here at all to not break the replication

Reply:
Is this even possible?